unitary/examples/quantum_chinese_chess/test_utils.py

```diff
@@ -12,14 +12,184 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 from unitary.alpha import QuantumObject, QuantumWorld
-from unitary.examples.quantum_chinese_chess.enums import SquareState
-from string import ascii_lowercase, digits
+from unitary.examples.quantum_chinese_chess.enums import SquareState, Type, Color
+from unitary.examples.quantum_chinese_chess.board import Board
+from unitary.examples.quantum_chinese_chess.piece import Piece
+from unitary import alpha
+from typing import List, Dict
+from collections import defaultdict
+from scipy.stats import chisquare
 
 
-# Build quantum objects a0 to i9, and add them to a quantum world.
-def init_board() -> QuantumWorld:
-    board = {}
-    for col in ascii_lowercase[:9]:
-        for row in digits:
-            board[col + row] = QuantumObject(col + row, SquareState.EMPTY)
-    return QuantumWorld(list(board.values()))
+_EMPTY_FEN = "9/9/9/9/9/9/9/9/9/9 w---1"
+
+
+def set_board(positions: List[str]) -> Board:
+    """Returns a board with the specified positions filled with
+    RED ROOKs.
+    """
+    board = Board.from_fen(_EMPTY_FEN)
+    for position in positions:
+        board.board[position].reset(
+            Piece(position, SquareState.OCCUPIED, Type.ROOK, Color.RED)
+        )
+        alpha.Flip()(board.board[position])
+    return board
+
+
+def location_to_bit(location: str) -> int:
+    """Transform location notation (e.g. "a3") into a bitboard bit number.
+    The return value ranges from 0 to 89.
+    """
+    x = ord(location[0]) - ord("a")
+    y = int(location[1])
+    return y * 9 + x
+
+
+def locations_to_bitboard(locations: List[str]) -> int:
+    """Transform a list of locations into a 90-bit board bitstring.
+    Each nonzero bit of the bitstring indicates that the corresponding
+    piece is occupied.
+    """
+    bitboard = 0
+    for location in locations:
+        bitboard += 1 << location_to_bit(location)
+    return bitboard
+
+
+def nth_bit_of(n: int, bit_board: int) -> bool:
+    """Returns the `n`-th (zero-based) bit of a 90-bit bitstring `bit_board`."""
+    return (bit_board >> n) % 2 == 1
+
+
+def bit_to_location(bit: int) -> str:
+    """Transform a bitboard bit number into a location (e.g. "a3")."""
+    y = bit // 9
+    x = chr(bit % 9 + ord("a"))
+    return f"{x}{y}"
+
+
+def bitboard_to_locations(bitboard: int) -> List[str]:
+    """Transform a 90-bit bitstring `bitboard` into a list of locations."""
+    locations = []
+    for n in range(90):
+        if nth_bit_of(n, bitboard):
+            locations.append(bit_to_location(n))
+    return locations
+
+
+def sample_board(board: Board, repetitions: int) -> List[int]:
```

Comment:
Optional:  should this be part of Board?

Reply:
Thanks Doug. It's moved to Board.